src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,318 @@
+    public void parseInput(String input) {
+        String[] inputs = input.split(" ");
+        String command = inputs[0];
+        try {
+            if (command.equals("list")) {
+                tasks.listTasks();
+            } else if (command.equals("delete")) {
+                parseDelete(input);
+            } else if (command.equals("mark")) {
+                parseMark(input);
+            } else if (command.equals("unmark")) {
+                parseUnMark(input);
+            } else if (command.equals("todo")) {
+                parseToDo(input);
+            } else if (command.equals("deadline")) {
+                parseDeadline(input);
+            } else if (command.equals("event")) {
+                parseEvent(input);
+            } else if (command.equals("find")) {
+                findTask(input);
+            } else {
+                throw new InvalidCommandException();
+            }
+        } catch (DukeException e) {
+            ui.showError(e.getMessage());
+        }
+    }
+
+    /**
+     * Handles the deleting of a specified task.
+     * Checks if a taskID is provided. Calls tasks.deleteTask(index) function to
+     * attempt task deletion.
+     *
+     * @param input user input.
+     * @throws NoTaskIdException      if no taskID is provided.
+     * @throws InvalidTaskIdException if a non-numerical id is provided.
+     */
+    public void parseDelete(String input) throws NoTaskIdException, InvalidTaskIdException {
+        String[] inputArr = input.split(" ");
+        if (inputArr.length == 1) {
+            throw new NoTaskIdException();
+        } else {
+            String strIndex = inputArr[1];
+            if (isNumber(strIndex)) {
+                int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
+                tasks.deleteTask(index);
+            } else {
+                //case where a number was not provided
+                throw new InvalidTaskIdException();
+            }
+        }
+    }
+
+    /**
+     * Handles the marking of a specified task.
+     * Checks if a taskID is provided. Calls the tasks.markTask(index) function to
+     * attempt task marking.
+     *
+     * @param input user input.
+     * @throws NoTaskIdException      if no taskID is provided.
+     * @throws InvalidTaskIdException If a non-numerical id is provided.
+     */
+    public void parseMark(String input) throws NoTaskIdException, InvalidTaskIdException {
+        String[] inputArr = input.split(" ", 2);
+        if (inputArr.length == 1) {
+            throw new NoTaskIdException();
+        } else {
+            String strIndex = inputArr[1];
+            if (isNumber(strIndex)) {
+                int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
+                if (tasks.isValidTaskId(index)) {
+                    tasks.markTask(index);
+                }
+            } else {
+                throw new InvalidTaskIdException();
+            }
+        }
```

Comment:
What happens if `tasks.isValidTaskId(index)` returns false? Could InvalidTaskIdException be thrown? I would write this part like this:
```
        if (inputArr.length == 1) {
            throw new NoTaskIdException();
        } else {
            String strIndex = inputArr[1];
            if (isNumber(strIndex) && tasks.isValidTaskId(Integer.parseInt(strIndex) - 1)) {
                int index = Integer.parseInt(strIndex) - 1; //because index starts from 1
                
                tasks.markTask(index);
                
            } else {
                throw new InvalidTaskIdException();
            }
        }
```